src/edu/buffalo/cse/jive/finiteStateMachine/monitor/Monitor.java

```diff
@@ -94,7 +96,9 @@ private String getEventKey(String value) {
 	 */
 	public boolean validate(List<Expression> expressions) throws Exception {
 		boolean result = validate(rootState, expressions);
-		rootState.setValid(result);
+		if(result && expressions.get(0) instanceof EExpression)rootState.setStatus(Status.MARKED);
+		else if(result)rootState.setStatus(Status.VALID);
+		else rootState.setStatus(Status.INVALID);

```

Comment:
If we have a **top-level valid EEexpression**, it sets the root state as MARKED, as that is the beginning of the graph.

The boolean values have also been replaced by enum.